src/main/java/bio/terra/cli/service/utils/GoogleCloudStorage.java

```diff
@@ -0,0 +1,81 @@
+package bio.terra.cli.service.utils;
+
+import bio.terra.cli.context.TerraUser;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility methods for calling Google Cloud Storage endpoints. */
+public class GoogleCloudStorage {
+  private static final Logger logger = LoggerFactory.getLogger(GoogleCloudStorage.class);
+
+  // the Terra user whose credentials will be used to call authenticated requests
+  private final TerraUser terraUser;
+
+  // the google project id of the workspace
+  private final String googleProjectId;
+
+  // the client object used for talking to Google Cloud Storage
+  private Storage storageClient;
```

Comment:
I think this should use CRL's StorageCow. Should be close to a drop-in replacement. https://github.com/DataBiosphere/terra-cloud-resource-lib/blob/master/google-storage/src/main/java/bio/terra/cloudres/google/storage/StorageCow.java

Reply:
What benefit would using CRL offer?

Reply:
Minimal at the moment, but consistent dev access to logging and utilities that we haven't yet built. @gjuggler  is CRL only for services or also worth using for the CLI?

Reply:
Ok, just noting here that this code will all be ripped out once WSM controlled resources are ready. However, there may still be some updates to controlled resources that we'd handle directly -- I'm not totally clear on that, yet.

Reply:
Good question re: the scope of CRL usage. I think we should use it wherever it's useful. In this case, I agree it's probably not worth adding the dependency for short-lived code. But if we do find ourselves repeatedly adding direct cloud platform interactions into the CLI, we should reassess.

Re: the CLI updating bucket policies directly, based on Dan's explorations of WSM controlled resources, current plans have all controlled bucket modifications needing to go through the WSM API, basically because permissions are not fine-grained enough. See [this section](https://docs.google.com/document/d/1F28KHeVpTh0Zuqdux30ONabMgjj6CNmyv7B8iqcXnsE/edit?resourcekey=0-5ndsiOmBJdG7bB1gZXOy1g#bookmark=id.au48n2159vom) for details.

Reply:
Thanks for the bookmark on bucket permissions. For other operations or types of controlled resources, we may end up talking directly to the cloud -- in that case, I'll look at using CRL.